scheduler.go

```diff
@@ -95,19 +95,22 @@ func (s *Scheduler) StartBlocking() {
 
 // StartAsync starts all jobs without blocking the current thread
 func (s *Scheduler) StartAsync() {
-	if !s.IsRunning() {
+	if s.running.CAS(false, true) {
```

Comment:
CAS is deprecated, docs say to use `CompareAndSwap`

Reply:
But, I wouldn't use that here. I'd leave it as `!s.IsRunning()`

Reply:
I didn't notice because go.mod depends on `go.uber.org/atomic v1.9.0` which does not yet have `CompareAndSwap`. 

There's an issue in `if !s.IsRunnin()`. Someone can start multiple goroutines calling this function and will probably get a race condition because it might call s.start() multiple times.

Reply:
It feels like settings s.running to true too early results in a start of multiple executors, so reverted that change.